kroxylicious-futures/src/main/java/io/kroxylicious/proxy/internal/future/PromiseInternal.java

```diff
@@ -10,13 +10,11 @@
  */
 package io.kroxylicious.proxy.internal.future;
 
-import io.netty.util.concurrent.FutureListener;
-
 import io.kroxylicious.proxy.future.Promise;
 
 /**
  * @author <a href="mailto:julien@julienviet.com">Julien Viet</a>
  */
-public interface PromiseInternal<T> extends Promise<T>, FutureListener<T>, FutureInternal<T> {
+public interface PromiseInternal<T> extends Promise<T>, FutureInternal<T> {
```

Comment:
Hmmm something a bit amis with the attribution here. 

Why do we have our own future package? What is it giving us over `j.u.c.Future`?

Reply:
It's a copy of `vertx` futures introduced here https://github.com/kroxylicious/kroxylicious/commit/30ef95e0df6f8330d9f761e965813238e503305d